Aplib.Core/Belief/MemoryBelief.cs

```diff
@@ -98,15 +98,12 @@ public override void UpdateBelief()
         /// <summary>
         /// Gets the memorized observation at a specific index.
         /// A higher index means a memory further back in time.
-        /// If the index is out of bounds, when clamped, returns the element closest to the index that is in bounds.
         /// </summary>
         /// <returns>The memory of the observation at the specified index.</returns>
-        public TObservation GetMemoryAt(int index, bool clamp = false)
+        public TObservation GetMemoryAt(int index)
         {
             int lastMemoryIndex = _memorizedObservations.Count;
-            if (clamp)
-                index = Math.Clamp(index, 0, lastMemoryIndex);
-            else if (index < 0 || index > lastMemoryIndex)
```

Comment:
Why did you remove this? I think it's alright to remove it, but then there must be a way of checking the current number of memorized observations. Right now there is no way of preventing an ArgumentOutOfRangeException, because we cannot check the allowed range of indices for the GetMemoryAt method.

Reply:
I thought you said in an earlier comment that the clamping was unwanted, but maybe I'm hallucinating, can't find it again. Will re-add